339.py

```diff
@@ -0,0 +1,63 @@
+# """
+# This is the interface that allows for creating nested lists.
+# You should not implement it, or speculate about its implementation
+# """
+#class NestedInteger:
+#    def __init__(self, value=None):
+#        """
+#        If value is not specified, initializes an empty list.
+#        Otherwise initializes a single integer equal to value.
+#        """
+#
+#    def isInteger(self):
+#        """
+#        @return True if this NestedInteger holds a single integer, rather than a nested list.
+#        :rtype bool
+#        """
+#
+#    def add(self, elem):
+#        """
+#        Set this NestedInteger to hold a nested list and adds a nested integer elem to it.
+#        :rtype void
+#        """
+#
+#    def setInteger(self, value):
+#        """
+#        Set this NestedInteger to hold a single integer equal to value.
+#        :rtype void
+#        """
+#
+#    def getInteger(self):
+#        """
+#        @return the single integer that this NestedInteger holds, if it holds a single integer
+#        The result is undefined if this NestedInteger holds a nested list
+#        :rtype int
+#        """
+#
+#    def getList(self):
+#        """
+#        @return the nested list that this NestedInteger holds, if it holds a nested list
+#        The result is undefined if this NestedInteger holds a single integer
+#        :rtype List[NestedInteger]
+#        """
+
+# Runtime: O(n + m)
+# n=num of elements
+# m = total depth of elements
+#
+# Space: O(n)
+# n = max depth
+class Solution:
+    def depthSum(self, nestedList: List[NestedInteger]) -> int:
+        total = 0
+        for x in nestedList:
+            total += self._depthSum(x, 1)
```

Comment:
This is very good to make a helper function that works on each NestedInteger rather than on the List[NestedInteger]